astropy/table/tests/test_item_access.py

```diff
@@ -251,6 +251,22 @@ def test_np_integers(self, table_data):
         t = table_data.Table(table_data.COLS)
         idxs = np.random.randint(len(t), size=2)
         item = t[idxs[1]]
+        
+    def test_np_scalar(self, table_data):
+        """
+        Select rows using numpy scalars.
+        """
+        t = table_data.Table(table_data.COLS)
+        idx = np.int64(np.random.randint(len(t)))
+        item = t[idx]
+        
+    def test_np_array_scalar(self, table_data):
+        """
+        Select rows using numpy array scalars.
+        """
+        t = table_data.Table(table_data.COLS)
+        idx = np.array(np.random.randint(len(t)))
+        item = t[idx]
```

Comment:
It would be more useful to check the result. It could also be a good idea to check that setting a value with a scalar index works.
I wonder if this test and `test_np_integers` could be merged with `test_row` using pytest's parametrize feature, with something like `@pytest.mark.parametrize("idx", [1, np.int64(1), np.array(1)])`.